runtime/sema/type_tags.go

```diff
@@ -935,8 +935,11 @@ func commonSuperTypeOfComposites(types []Type) Type {
 	}
 
 	if hasCommonInterface {
+		if len(commonInterfacesList) == 0 {
+			panic(errors.NewUnreachableError())
+		}
```

Comment:
this is rather unnecessary, because `hasCommonInterface` is set based on the length of `commonInterfaces`/`commonInterfacesList`